src/middlewared/middlewared/plugins/vm/vms.py

```diff
@@ -101,10 +101,18 @@ async def do_create(self, data):
         `shutdown_timeout` indicates the time in seconds the system waits for the VM to cleanly shutdown. During system
         shutdown, if the VM hasn't exited after a hardware shutdown signal has been sent by the system within
         `shutdown_timeout` seconds, system initiates poweroff for the VM to stop it.
+
+        SCALE Angelfish: Specifying `devices` is deprecated and will be removed in next major release.
         """
         async with LIBVIRT_LOCK:
             await self.middleware.run_in_thread(self._check_setup_connection)
 
+        if data.get('devices'):
+            self.logger.warning(
+                'SCALE Angelfish: Specifying "devices" in "vm.create" is deprecated and will be '
+                'removed in next major release.'
+            )
+
```

Comment:
Thats usually done in python using warnings module, wonder if it makes sense to use that too